### PR TITLE
some overhaul

### PR DESCRIPTION
**Pull request by gregor-i**
* added a option to surpress postgres output
* added utility to chose a port at random
* preventing multiple postgres instances
* updated sbt examples

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -20,17 +20,25 @@ To use the postgresql in your project have your tests depend on starting the Pos
...
 lazy val root = (project in file(".")).enablePlugins(EmbeddedPostgresPlugin)
 
-testOptions in Test <+= postgresTestCleanup // clean up postgresql after tests
+testOptions in Test += postgresTestCleanup.value // clean up postgresql after tests
 
-test in Test <<= (test in Test).dependsOn(startPostgres)
+test in Test := (test in Test).dependsOn(startPostgres)
 
-testOnly in Test <<= (testOnly in Test).dependsOn(startPostgres)
+testOnly in Test := (testOnly in Test).dependsOn(startPostgres)
```
Comment (geekity) [inline README.md]: When run with `sbt.version=0.13.11` the above `build.sbt` config changes didn't work right. Running `sbt test` just produces following output without any tests
```
[info] Loading global plugins from /Users/foo/.sbt/0.13/plugins
[info] Loading project definition from /Users/foo/path/to/my-project/project
[info] Set current project to my-project (in build file:/Users/foo/path/to/my-project)
[success] Total time: 0 s, completed 09-Oct-2017 15:44:22
```

**Review by geekity — CHANGES_REQUESTED** [review]
Really good change overall, just the start up doc change doesn't work when I tried it out.

Thanks for the contribution! :)

File: README.md (hunk 1 above)
Comment (geekity) [reply README.md]: The configs in question:

```
// these worked when I tried them:
// testOptions in Test <+= postgresTestCleanup
// test in Test <<= (test in Test).dependsOn(startPostgres)
// testOnly in Test <<= (testOnly in Test).dependsOn(startPostgres)

// these changes didn't run for me:
testOptions in Test += postgresTestCleanup.value
test in Test := (test in Test).dependsOn(startPostgres)
testOnly in Test := (testOnly in Test).dependsOn(startPostgres)
```

Comment (gregor-i) [reply README.md]: Yeah, you're right. The documentation is intended to be used with `sbt.version=0.13.16`.

How do you want me to address this issue? I could:
* revert the documentation to be used with your sbt-version and ignore the deprecation warnings with higher sbt versions.
* add documentation for old and new sbt versions.

edit: I just reverted the docu examples.

Comment (geekity) [reply README.md]: I'm happy with either of the 2 suggestions, so either one that suits you better :)

**Review by geekity — APPROVED** [review]

**Comment by geekity** [comment]
👍  If you are happy with the changes, I am happy to merge this in

**Comment by gregor-i** [comment]
I'm very happy with the changes. As soon as the changes are merged and the jar is published, I can use your plugin in my code :)

**Comment by geekity** [comment]
@gregor-i Built and published, seems to be working fine on my end

**Pull request merged**